tests/rptest/infinite_retention/infinite_retention_test.py

```diff
@@ -0,0 +1,794 @@
+# Copyright 2023 Redpanda Data, Inc.
```

Comment:
This test is defined outside of the existing `tests/` and `scale_tests/` paths -- those paths act as the entry points for CI jobs that execute tests, how is this one going to be executed?

You might find this fits inside `scale_tests/` the tests that are run on dedicated nodes and expected to have substantial I/O and runtime.